### ISSUE TITLE
Callback Error

### ISSUE THREAD

**Issue by stn1978** [haydenbleasel/favicons#27]
Hi Hayden!

I've got an error in my configuration an I'd like to share it with you. Maybe you have a solution?

The generator works but I keep getting the following error message:

```javascript

/Users/user/Projects/project/node_modules/favicons/index.js:152
                return callback(metadata);
                       ^
TypeError: undefined is not a function
    at Object.rfg.callback (/Users/user/Projects/project/node_modules/favicons/index.js:152:24)
    at /Users/user/Projects/project/node_modules/favicons/node_modules/real-favicon/index.js:121:57
    at /Users/user/Projects/project/node_modules/favicons/node_modules/real-favicon/index.js:116:28
    at done (/Users/user/Projects/project/node_modules/favicons/node_modules/real-favicon/node_modules/async/lib/async.js:135:19)
    at /Users/user/Projects/project/node_modules/favicons/node_modules/real-favicon/node_modules/async/lib/async.js:32:16
    at /Users/user/Projects/project/node_modules/favicons/node_modules/real-favicon/index.js:113:25
    at /Users/user/Projects/project/node_modules/favicons/node_modules/real-favicon/index.js:32:25
    at /Users/user/Projects/project/node_modules/gulp-replace/node_modules/istextorbinary/node_modules/safefs/node_modules/graceful-fs/graceful-fs.js:104:5
    at Object.oncomplete (fs.js:107:15)

```

That's my configuration:

```javascript
gulp.task('favicons', function () {
	favicons({
		files: {
			src:  'src/assets/images/favicon-src.png',
			dest: 'dist/assets/images/favicons',
			html: 'dist/favicons.html'
		},
		icons: {
			android:      true,
			appleIcon:    true,
			appleStartup: true,
			coast:        true,
			favicons:     true,
			firefox:      true,
			opengraph:    true,
			windows:      true,
			yandex:       true
		},
		settings: {
			appName:        'test.com',
			appDescription: 'Personal Website of Name',
			developer:      'Name',
			developerURL:   'http://test.com/',
			version:        0.1,
			background:     '#ffffff',
			index:          'dist/index.html',
			url:            'http://test.com/',
			silhouette:     true,
			logging:        false
		}
	});
});
```

What am I doing wrong? What can I do?

**Comment by haydenbleasel** [comment]
This is actually my fault for not checking if a callback exists - I just fixed it then for favicons-2.1.1.

Anyway, you could fix this issue by specifying a callback function OR (much easier) use [gulp-favicons](https://www.npmjs.com/package/gulp-favicons).

**Comment by stn1978** [comment]
Thanks! Yes, fixed in 2.1.1. :)
